pkg/pool-gyro/contracts/Gyro2CLPPool.sol

```diff
@@ -167,7 +168,7 @@ contract Gyro2CLPPool is IGyro2CLPPool, BalancerPoolToken {
 
         uint256 currentInvariant = Gyro2CLPMath.calculateInvariant(balances, sqrtAlpha, sqrtBeta, rounding);
 
-        uint256[2] memory virtualOffsets = _calculateVirtualOffsets(currentInvariant, sqrtAlpha, sqrtBeta);
+        uint256[2] memory virtualOffsets = _calculateVirtualOffsets(currentInvariant, sqrtAlpha, sqrtBeta, rounding);
```

Comment:
I have some doubts about these.

We're rounding both in the same direction, but then those are used inside swap for `calcInGivenOut` and `calcOutGivenIn`, which need to round the result differently.

Reply:
Fixed!